File: skyspacefivefiftyfive_ui/src/services/AgencyBookingService.js
import http from "../http-common"; 

class AgencyBookingService {
  getAllAgencyBookings(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/agencyBooking/agencyBookings`, searchDTO);
  }

  get(agencyBookingId) {
    return this.getRequest(`/agencyBooking/${agencyBookingId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/agencyBooking?field=${matchData}`, null);
  }

  addAgencyBooking(data) {
    return http.post("/agencyBooking/addAgencyBooking", data);
  }

  update(data) {
  	return http.post("/agencyBooking/updateAgencyBooking", data);
  }
  
  uploadImage(data,agencyBookingId) {
  	return http.postForm("/agencyBooking/uploadImage/"+agencyBookingId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AgencyBookingService();

File: src/main/java/com/skyspacefivefiftyfive/service/AgencyBookingService.java
package com.skyspacefivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefivefiftyfive.domain.AgencyBooking;
import com.skyspacefivefiftyfive.dto.AgencyBookingDTO;
import com.skyspacefivefiftyfive.dto.AgencyBookingSearchDTO;
import com.skyspacefivefiftyfive.dto.AgencyBookingPageDTO;
import com.skyspacefivefiftyfive.dto.AgencyBookingConvertCriteriaDTO;
import com.skyspacefivefiftyfive.service.GenericService;
import com.skyspacefivefiftyfive.dto.common.RequestDTO;
import com.skyspacefivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AgencyBookingService extends GenericService<AgencyBooking, Integer> {

	List<AgencyBooking> findAll();

	ResultDTO addAgencyBooking(AgencyBookingDTO agencyBookingDTO, RequestDTO requestDTO);

	ResultDTO updateAgencyBooking(AgencyBookingDTO agencyBookingDTO, RequestDTO requestDTO);

    Page<AgencyBooking> getAllAgencyBookings(Pageable pageable);

    Page<AgencyBooking> getAllAgencyBookings(Specification<AgencyBooking> spec, Pageable pageable);

	ResponseEntity<AgencyBookingPageDTO> getAgencyBookings(AgencyBookingSearchDTO agencyBookingSearchDTO);
	
	List<AgencyBookingDTO> convertAgencyBookingsToAgencyBookingDTOs(List<AgencyBooking> agencyBookings, AgencyBookingConvertCriteriaDTO convertCriteria);

	AgencyBookingDTO getAgencyBookingDTOById(Integer agencyBookingId);







}





